FaroShuffle/FaroShuffle/Extensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace FaroShuffle
+{
+    public static class Extensions
+    {
+        public static IEnumerable<T> InterleaveSequenceWith<T>(this IEnumerable<T> first, IEnumerable<T> second)
+        {
+            var firstIter = first.GetEnumerator();
+            var secondIter = second.GetEnumerator();
+
+            while (firstIter.MoveNext()  && secondIter.MoveNext())  // note for self - enumerator starts before 1st element
+                                                                    // must call MoveNext to advance to first element before reading current
+            {
+                yield return firstIter.Current;
+                yield return secondIter.Current;
+            }
+        }
+
+        public static bool SequenceEquals<T>(this IEnumerable<T> first, IEnumerable<T> second)
```

Comment:
Comments

Also, wouldn't it be more effective to override the = operator than add this strange method SequenceEquals?